fapi-client/src/main/scala/com/gu/facia/api/utils/ResolvedMetaData.scala

```diff
@@ -0,0 +1,99 @@
+    showKickerCustom = false,
+    showBoostedHeadline = false,
+    showMainVideo = false,
+    showKickerTag = false,
+    showByline = false,
+    imageCutoutReplace = false,
+    showQuotedHeadline = false)
+
+  private[utils] def fromTrailMetaData(trailMeta: MetaDataCommonFields): ResolvedMetaData =
+    ResolvedMetaData(
+      isBreaking = trailMeta.isBreaking.exists(identity),
+      isBoosted = trailMeta.isBoosted.exists(identity),
+      imageHide = trailMeta.imageHide.exists(identity),
+      imageReplace = trailMeta.imageReplace.exists(identity),
+      showKickerSection = trailMeta.showKickerSection.exists(identity),
+      showKickerCustom = trailMeta.showKickerCustom.exists(identity),
+      showBoostedHeadline = trailMeta.showBoostedHeadline.exists(identity),
+      showMainVideo = trailMeta.showMainVideo.exists(identity),
+      showKickerTag = trailMeta.showKickerTag.exists(identity),
+      showByline = trailMeta.showByline.exists(identity),
+      imageCutoutReplace = trailMeta.imageCutoutReplace.exists(identity),
+      showQuotedHeadline = trailMeta.showQuotedHeadline.exists(identity))
+
+  private[utils] def fromContent(content: Content): ResolvedMetaData = {
+    val mutatingMetaDataDefaults = Default
+    if (isCartoonForContent(content))
+      mutatingMetaDataDefaults.copy(showByline = true)
+    else if (isCommentForContent(content))
+      mutatingMetaDataDefaults.copy(
+        showByline = true,
+        showQuotedHeadline = true,
+        imageCutoutReplace = true)
+    else if (isVideoForContent(content))
+      mutatingMetaDataDefaults.copy(showMainVideo = true)
+    else
+      mutatingMetaDataDefaults}
+
+  private [utils] def joinResolvedMetaDataAndTrailMetaData(first: ResolvedMetaData, second: MetaDataCommonFields): ResolvedMetaData =
+    first.copy(
+      isBreaking = second.isBreaking.getOrElse(first.isBreaking),
+      isBoosted = second.isBoosted.getOrElse(first.isBoosted),
+      imageHide = second.imageHide.getOrElse(first.imageHide),
+      imageReplace = second.imageReplace.getOrElse(first.imageReplace),
+      showKickerSection = second.showKickerSection.getOrElse(first.showKickerSection),
+      showKickerCustom = second.showKickerCustom.getOrElse(first.showKickerCustom),
+      showBoostedHeadline = second.showBoostedHeadline.getOrElse(first.showBoostedHeadline),
+      showMainVideo = second.showMainVideo.getOrElse(first.showMainVideo),
+      showKickerTag = second.showKickerTag.getOrElse(first.showKickerTag),
+      showByline = second.showByline.getOrElse(first.showByline),
+      imageCutoutReplace = second.imageCutoutReplace.getOrElse(first.imageCutoutReplace),
+      showQuotedHeadline = second.showQuotedHeadline.getOrElse(first.showQuotedHeadline)
+    )
+
+  def fromContentAndTrailMetaData(content: Content, trailMeta: MetaDataCommonFields): ResolvedMetaData = {
+    val metaDataFromContent = fromContent(content)
+    joinResolvedMetaDataAndTrailMetaData(metaDataFromContent, trailMeta)
```

Comment:
Because it initially worked differently, and I left it the way it is. Could just inline it I suppose.